model/vfs/couchdb_indexer.go

```diff
@@ -226,10 +226,19 @@ func (c *couchdbIndexer) moveDir(oldpath, newpath string) error {
 		return nil
 	}
 
-	// We limit the stack to 1024 bulk updates to avoid infinite loops, as we
+	// We limit the stack to 128 bulk updates to avoid infinite loops, as we
 	// had a case in the past.
-	for i := 0; i < 1024; i++ {
-		sel := mango.StartWith("path", oldpath+"/")
+	start := oldpath + "/"
+	for i := 0; i < 128; i++ {
+		// The simple selector mango.StartWith can have some issues when
+		// renaming a folder to the same name, but with a different unicode
+		// normalization (like NFC to NFD). In that case, CouchDB would always
+		// return the same documents with this selector, as it does the
+		// comparison on a normalized string.
+		sel := mango.And(
+			mango.Gt("path", start),
+			mango.Lt("path", oldpath+"0"), // 0 is the next ascii character after /
```

Comment:
This could be assigned to an `end` variable next to `start`, don't you think?

Reply:
What happens if we have the following directories:
- `folder1/`
- `folder10/`

Will `folder10/` be selected?

Reply:
It won't be selected: `Lt` means `<`, it is Lte for `<=`.

Reply:
Why do you want to use a variable for `end`?

Reply:
OK, it makes sense for `Lt`.
I find the `path+"0"` value a bit magical and thought it would deserve to be documented apart from where it's used. And besides, the `start` variable was extracted but I see that's because you're re-using it later.

It can stay this way if you prefer.